main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala

```diff
@@ -287,6 +292,36 @@ class LanguageServer(port: Int) extends WebSocketServer(new InetSocketAddress("l
 
   }
 
+  private def sourceWithHole(source: String) = {
+    val lines = source.linesWithSeparators
+    val (prefix, rest) = lines.splitAt(mostRecentPosition.line - 1)
+    val line = rest.next()
+    val (linePrefix, lineRest) = line.splitAt(mostRecentPosition.character - 1)
+    (prefix ++ (linePrefix + " ??? " + lineRest) ++ rest).mkString
+  }
+
+  private def checkSource(): Validation[TypedAst.Root, CompilationMessage] = {
+    val result = flix.check()
+    result match {
+      case r: Success[_, _] => r
+
+      case Failure(errors) =>
+        val loc = errors.head.loc
```

Comment:
Plausibly this should only be done if there is exactly 1 error?

Reply:
I'm not sure I see any benefit in this. In many languages (perhaps not in Flix, I don't have enough experience to be sure) a single typo can result in many subsequent errors. If adding `???` turns (say) 3 errors into zero errors, it's a pretty good guess that that's the case.

And if it turns 3 errors into 2, then we drop the result of the compilation on the floor, so no harm done.